trollmoves/server.py

```diff
@@ -702,29 +767,40 @@ def process_IN_DELETE(self, event):
                 try:
                     self._watchManager.rm_watch(self._watched_dirs[event.pathname], quiet=False)
                 except pyinotify.WatchManagerError:
-                    # As the directory is deleted prior removing the watch will cause a error message
-                    # from pyinotify. This is ok, so just pass the exception.
+                    # As the directory is deleted prior removing the
+                    # watch will cause a error message from
+                    # pyinotify. This is ok, so just pass the
+                    # exception.
                     pass
                 finally:
                     del self._watched_dirs[event.pathname]
             except KeyError:
-                LOGGER.warning("Dir {} not watched by inotify. Can not delete watch.".format(event.pathname))
+                LOGGER.warning(
+                    "Dir %s not watched by inotify. Can not delete watch.",
+                    event.pathname)
         return
 
 
-def process_old_files(pattern, fun):
+def process_old_files(pattern, fun, publisher, kwargs):
+    """Process files from *pattern* with function *fun*."""
     fnames = glob.glob(pattern)
     if fnames:
         # time.sleep(3)
         LOGGER.debug("Touching old files")
         for fname in fnames:
             if os.path.exists(fname):
-                fun(fname)
+                fun(fname, publisher, pattern, kwargs)
```

Comment:
`create_inotify_notifier` return a function that only takes a filename, how can this work ?

Reply:
The rest of the arguments were forgotten from the `fun()` definition, the call it was doing had them.